### ISSUE TITLE
Support for writing a non-spatial table to a GeoPackage

### ISSUE THREAD

**Issue by florisvdh** [r-spatial/sf#1345]
In [its layer options](https://gdal.org/drivers/vector/gpkg.html#layer-creation-options), GDAL's GPKG-vector driver offers the possibility to store non-spatial attribute tables (~dataframes), using the option `ASPATIAL_VARIANT`. However it seems that `st_write()` does not support this:

```r
st_write(nonspatial_df,
         "existing_gpkg.gpkg"), 
         layer = "nonspatial_table1",
         driver = "GPKG",
         layer_options = "ASPATIAL_VARIANT=GPKG_ATTRIBUTES")
```

This throws an error:

```
Error in st_sf(x, ..., agr = agr, sf_column_name = sf_column_name) : 
  no simple features geometry column present
```

Is there an alternative to properly use this GDAL feature from R, or else could it be incorporated in `st_write()`?

An alternative is to manually insert the table with `RSQLite::dbWriteTable()`, which does work with `st_read()` afterwards - probably because GDAL [lists all non-spatial tables](https://gdal.org/drivers/vector/gpkg.html#non-spatial-tables) even when they were not registered in any GeoPackage system tables. However this approach of writing violates the GPKG standard, which e.g. requires registration of tables in the `gpkg_contents` table.

**Comment by edzer** [comment]
Incorporating this in `st_write` would be quite a lot of work, and is not something that I would do voluntarily.

**Comment by edzer** [comment]
Closed by #1396.
